app/models/operatingsystem.rb

```diff
@@ -246,10 +246,10 @@ def boot_filename(host = nil)
     return default_boot_filename if host.nil? || host.pxe_loader.nil?
     return host.foreman_url('iPXE') if host.pxe_loader == 'iPXE Embedded'
     architecture = host.arch.nil? ? '' : host.arch.bootfilename_efi
-    if host.subnet&.httpboot?
+    if host.subnet&.httpboot? && host.pxe_loader =~ /UEFI HTTP/
       if host.pxe_loader =~ /UEFI HTTPS/
         port = host.subnet.httpboot.httpboot_https_port!
-      else
+      elsif host.pxe_loader =~ /UEFI HTTP/
```

Comment:
Can't this stay just `else`? We already know `host.pxe_loader` matches `/UEFI HTTP/` from line 249

Reply:
Sure, rebased.